nestkernel/recording_device.cpp

```diff
@@ -595,12 +642,30 @@ nest::RecordingDevice::record_event( const Event& event, bool endrecord )
   const Time stamp = event.get_stamp();
   const double offset = event.get_offset();
   const double weight = event.get_weight();
+  const long port = event.get_port();
+  const long rport = event.get_rport();
 
-  // std::cout << "recording device sender: " << sender << std::endl;
+  index receiver = -1;
+  if ( P_.withreceivergid_ )
+  {
+    const WeightRecorderEvent* wr_e =
+      dynamic_cast< const WeightRecorderEvent* >( &event );
+    if ( wr_e != 0 )
+    {
+      receiver = wr_e->get_receiver_gid();
+    }
+    else
+    {
+      receiver = event.get_receiver_gid();
+    }
+  }
```

Comment:
I think you are manually doing run-type type resolution here. `e.get_receiver_gid()` should be all that is needed. So you could do
```C++
const long receiver = P_.withtarget_ ? e.get_receiver_gid() : -1;
```
(if it is at all necessary to use the dummy -1, why not pass the target GID in any case?)